.github/workflows/ci.yml

```
@@ -197,123 +190,153 @@ jobs:
-          curl -sL "https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr22.fa.gz" \
+          curl --fail --retry 3 --retry-delay 2 --retry-all-errors -sSL \
+            "https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr22.fa.gz" \
             | gunzip > /tmp/chr22/chr22.fa
           test -s /tmp/chr22/chr22.fa
           echo "chr22.fa bytes: $(wc -c < /tmp/chr22/chr22.fa)"
 
       - name: Build chr22 FMI index
-        if: matrix.canonical == true
+        # bwa-mem3 index is rebuilt every run because its sidecar layout is
+        # coupled to the bwa-mem3 binary, which changes per commit.
         run: |
```

actionlint (1.7.12)

[error] shellcheck reported issue in this script: SC2086:info:2:1: Double quote to prevent globbing and word splitting

(shellcheck)